qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -228,34 +260,52 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
             if self._sample_from_iterations:
                 power = np.random.randint(power)
             # Run a grover experiment for a given power of the Grover operator.
-            if self._quantum_instance.is_statevector:
-                qc = self.construct_circuit(amplification_problem, power, measurement=False)
-                circuit_results = self._quantum_instance.execute(qc).get_statevector()
+            if self._sampler:
```

Comment:
```suggestion
            if self._sampler is not None:
```
Better when checking objects for None to do so explicitly since they can override bool so it maybe a valid object yet this might be False.